plc4j/api/src/main/java/org/apache/plc4x/java/api/messages/PlcFieldResponse.java

```diff
@@ -0,0 +1,42 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+package org.apache.plc4x.java.api.messages;
+
+import org.apache.plc4x.java.api.model.PlcField;
+import org.apache.plc4x.java.api.types.PlcResponseCode;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Base type for all response messages sent as response for a prior request
+ * from a plc to the plc4x system.
+ * @param <REQUEST_TYPE>
+ */
+public interface PlcFieldResponse<REQUEST_TYPE extends PlcFieldRequest> extends PlcResponse<REQUEST_TYPE> {
+
+    Map<String, PlcField> getFields();
+
+    Collection<String> getFieldNames();
```

Comment:
This should be more a `java.util.Set` as these are unique, aren't they?

Reply:
Well yes and no ... this hasn't been finalized, but the order of the items it important and Set isn't ordered but Collection isn't either, so I'll have to change that to something else anyway.